src/methods.cpp

```diff
@@ -3321,15 +3321,15 @@ PyObject* meth_get_phy_firmware_version(PyObject* self, PyObject* args)
 			return set_ics_exception(exception_runtime_error(), dll_get_error(buffer));
 		}
 
-		ice::Function<int __stdcall (ICS_HANDLE, char, unsigned int*, unsigned int*)> icsneoGetPhyFwVersion(lib, "icsneoGetPhyFwVersion");
+		ice::Function<int __stdcall (ICS_HANDLE, char, int*, int*)> icsneoGetPhyFwVersion(lib, "icsneoGetPhyFwVersion");
 
         Py_BEGIN_ALLOW_THREADS
-            if (!icsneoGetPhyFwVersion(handle, phy_indx, &phyVersion, &functionError)) {
+            if (!icsneoGetPhyFwVersion(handle, phy_indx, &phy_version, &function_error)) {
```

Comment:
we should be checking against `function_error` and raising an exception if its not valid.